src/components/checkbox/checkbox.spec.ts

```diff
@@ -0,0 +1,749 @@
+        fixture.detectChanges();
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('aria-disabled')).toEqual('false');
+      }).then(done).catch(done);
+    });
+
+    it('adds a default tabindex of 0 to the host element', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('tabindex')).toEqual('0');
+      }).then(done).catch(done);
+    });
+
+    it('allows clients to provide their own tabindex', function(done: () => void) {
+      builder.createAsync(CheckboxCustomTabindexController).then(function(fixture) {
+        fixture.detectChanges();
+        const component = fixture.componentInstance;
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('tabindex')).toEqual(
+            String(component.checkboxTabindex));
+      }).then(done).catch(done);
+    });
+
+    it('allows clients to provide an aria-label', function(done: () => void) {
+      builder.createAsync(CheckboxCustomArialabelController).then(function(fixture) {
+        fixture.detectChanges();
+        const component = fixture.componentInstance;
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('aria-label')).toEqual(component.checkboxLabel);
+      }).then(done).catch(done);
+    });
+
+    it('sets the "aria-labelledby" attribute to the id of the label', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        const el = fixture.debugElement.query(By.css('.mc-checkbox'));
+        const label = el.nativeElement.querySelector('label');
+        expect(el.nativeElement.getAttribute('aria-labelledby')).toEqual(label.id);
+      }).then(done).catch(done);
+    });
+
+    describe('when given an "align" input with a value of "end"', function() {
+      var fixture: ComponentFixture;
+
+      beforeEach(function(done: () => void) {
+        builder.createAsync(CheckboxEndAlignedController).then(function(_fixture_) {
+          fixture = _fixture_;
+          fixture.detectChanges();
+        }).then(done).catch(done);
+      });
+
+      it('sets an "md-checkbox-align-end" class on the host element', function() {
+        const el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.className).toContain('md-checkbox-align-end');
+      });
+    });
+
+    describe('when the checkbox\'s checked value is set', function() {
+      var fixture: ComponentFixture;
+      var controller: CheckboxController;
+      var changePromise: Promise<boolean>;
+      var waitingForChange: boolean;
+
+      function waitForChange(): Promise<boolean> {
```

Comment:
You inject `NgZone` the same was as `TestComponentBuilder`:
```ts
inject([TestComponentBuilder, NgZone], (tcb, ngZone) => {
  zone = ngZone;
});
```

Then, in the test, you could try
```
zone.onEventDone(() => {
  // make assertions
});
```

I'm not sure if it will actually work, but it may reduce the code you have here without going to `fakeAsync` and its problems with `templateUrl`.